backend/src/test/factory/DiscountFactory.ts

```diff
@@ -0,0 +1,11 @@
+import { DiscountType } from '../../main/data/DiscountType';
+import { Discount } from '../../main/entity/Discount';
+
+export class DiscountFactory {
+  generate(discountType?: DiscountType, discountValue?: number): Discount {
+    return new Discount(
+      discountType ?? DiscountType.AMOUNT,
+      discountValue ?? 188.08
```

Comment:
By default for this here and for the other factory classes, we're setting defaults? So 188.08 would be the discountValue if the parameter is null/undefined?

Reply:
How come we're doing it in certain places too? 😅  Sorry, not too familiar with this!

Reply:
Yup those are default values since the parameters we defined inside `generate` are optional because maybe in testing we don't care about setting certain fields and we leave it to the factory to randomly generate this. 

But if you take a look at `PromotionFactory` you'll see that `user, schedules, discount` are not optional values, because those are mandatory fields we want to enforce when testing

Reply:
However, if we wanted to automatically add `user, schedules, discount` when we use the Factory method, we could create another function like this:

```
export class PromotionFactory {
  generateWithUserDiscountAndSchedule( <other parameters>
  ): Promotion {
    return new Promotion(
      new UserFactory().generate(),
      new DiscountFactory().generate(),
      [new ScheduleFactory().generate()]
      ... <other parameters>
    );
  }
}
```

Reply:
I see!! Is there a harm to keeping the optional values as null/undefined? I'm just thinking if it might be confusing when we're checking test results or debugging!

Reply:
Yeah because we have some non-nullable columns, so we can't persist them if they're undefined

Reply:
But yeah it should be pretty okay like I don't see us writing too many tests that concern some of the undefined fields in the factory methods. And if it is confusing for the person who's debugging or writing the tests, what's nice is that you can provide your own values to the factory methods if you want.

Reply:
Hmm, so even though we're only providing these default values for optional variables, you're saying that some of these can't be null/undefined?

Reply:
Yeah like these optional parameters are non-nullable in the DB. I don't think we can persist undefined, but regardless like I think generating these random values won't do harm because if it's confusing we can just override them!
![image](https://user-images.githubusercontent.com/49849754/103240776-1cfe6e80-4906-11eb-835e-e6021812bbfb.png)

Reply:
Oooooh okay! As long as we remember with our tests in the future, but sounds good!